sdk/core/core/CMakeLists.txt

```diff
@@ -51,26 +51,28 @@ target_include_directories (az_core PUBLIC inc)
 # make sure that users can consume the project as a library.
 add_library (az::core ALIAS az_core)
 
-add_executable (az_core_test test/main.c)
```

Comment:
I'd recommend putting this branch outside this directory, just only add the sdk/core/core directory if it's on.

Reply:
oh it's already outside. Given that this directory won't be part of the build unless the BUILD_AZ_CORE option is ON don't think you need the if(BUILD_AZ_CORE) in this file...

Reply:
`BUILD_AZ_CORE` is ON by default.  See line 55, there is where we check if BUILD_AZ_CORE is ON for creating test executable and other integration tests poc.

Reply:
Right but you ALREADY check for BUILD_AZ_CORE == ON in the outer cmakelists file, so the check on line 55 is redundant (sorry this comment is off by a few lines :D)